lib/tanker/crypto.rb

```diff
@@ -7,12 +7,22 @@ module Crypto
     class InvalidSignature < StandardError; end
 
     HASH_MIN_SIZE = RbNaCl::Hash::Blake2b::BYTES_MIN # 16
+    BLOCK_HASH_SIZE = 32
 
     def self.generichash(input, size)
       binary_input = input.dup.force_encoding(Encoding::ASCII_8BIT)
       RbNaCl::Hash.blake2b(binary_input, digest_size: size)
     end
 
+    def self.hash_user_id(app_id, user_id)
+      binary_user_id = user_id.dup.force_encoding(Encoding::ASCII_8BIT)
```

Comment:
Aren't you afraid of Unicode?

Reply:
This is actually not a string, but a `char*` pointing to binary data. Ruby does not have a binary type, so they always use ASCII_8BIT to represent binary.

This is also what encrypt/decrypt take and return.

We have to dup because force_encoding changes the input representation IIRC, so without the dup we would be modifying `user_id` here. (note that all this code is taken from sdk-ruby, where we first decided how to handle binary)